### PR TITLE
[feature/fileprovider-passcode-UI] File Provider Passcode UI 

### PR DESCRIPTION
**Pull request by hosy**
## Description
Added a passcode UI for the file provider, when a passcode was set in the app. With this UI the file provider can be used with passcode set immediately. Passcode can be entered directly in the file provider extension.

In addition the resign script and fastlane script got an update to set all identifiers and provisioning profiles.

## Related Issue
#880 

## Motivation and Context
Secure the app with passcode set immediately and and use the file provider extension by unlocking the passcode in the extension.

## How Has This Been Tested?
- Set a passcode
- Lock application `immediately`
- Open Files.app
- Select ownCloud file provider
- Tap `Authorize`
- Passcode UI should appear and disappear after entering the correct passcode
- Check Biometrical unlock
- Check file provider unlock in third party apps

## Screenshots (if appropriate):

https://user-images.githubusercontent.com/736109/106192989-374d9580-61ad-11eb-97cf-22f1db43e1d9.mov

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING**](https://github.com/owncloud/ios-app/blob/master/CONTRIBUTING.md) document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

## QA

Checks: https://github.com/owncloud/ios-app/pull/881#issuecomment-795038786

Bugs & Improvements:

- [X] (1) Passcode view always dark https://github.com/owncloud/ios-app/pull/881#issuecomment-795040139
- [X] (2) Brute force protection https://github.com/owncloud/ios-app/pull/881#issuecomment-795049508
- [x] (3) App is displayed after typing correct passcode in Files app (iPad) https://github.com/owncloud/ios-app/pull/881#issuecomment-797480195
- [X] (4) Touch id does not unblock files app https://github.com/owncloud/ios-app/pull/881#issuecomment-799300538

### CODE REVIEW

**Review by felix-schwarz — CHANGES_REQUESTED** [review]

File: ownCloud File Provider UI/DocumentActionViewController.swift [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+			}
+
+			AppLockManager.shared.showLockscreenIfNeeded()
+		} else {
+			let collection = Theme.shared.activeCollection
+			self.view.backgroundColor = collection.toolbarColors.backgroundColor
+			label.textColor = collection.toolbarColors.labelColor
+			button.setTitleColor(collection.toolbarColors.labelColor, for: .normal)
+			button.backgroundColor = collection.neutralColors.normal.background
+			label.text = "Passcode protection is not supported on this device.\nPlease disable passcode lock in the app settings.".localized
```
[the remark this answers is not in the document]
Comment (hosy) [reply ownCloud File Provider UI/DocumentActionViewController.swift:46]: Fixed!

File: ownCloud/Settings/SecuritySettingsSection.swift [hunk 2]
```diff
@@ -104,7 +104,7 @@ class SecuritySettingsSection: SettingsSection {
 
 				let newVC = StaticTableViewController(style: .grouped)
 				newVC.title = "Lock application".localized
-				let frequencySection = StaticTableViewSection(headerTitle: "Lock application".localized, footerTitle: "If you choose \"Immediately\" the App will be locked, when it is no longer in foreground.\n\nAccess in Files.app is not possible, if you choose lock interval \"Immediately\".\nPlease choose an other delay, if you want to access your files in the Files.app or via a document picker.".localized)
+				let frequencySection = StaticTableViewSection(headerTitle: "Lock application".localized, footerTitle: "If you choose \"Immediately\" the App will be locked, when it is no longer in foreground.".localized)
```
[the remark this answers is not in the document]
Comment (hosy) [reply ownCloud/Settings/SecuritySettingsSection.swift:107]: Fixed!

File: ownCloud File Provider UI/Info.plist [hunk 3]
```diff
@@ -5,7 +5,7 @@
 	<key>CFBundleDevelopmentRegion</key>
 	<string>$(DEVELOPMENT_LANGUAGE)</string>
 	<key>CFBundleDisplayName</key>
-	<string>$(PRODUCT_NAME) File ProviderUI</string>
+	<string>ownCloud</string>
```
[the remark this answers is not in the document]
Comment (hosy) [reply ownCloud File Provider UI/Info.plist]: Fixed!

**Comment by hosy** [comment]
@felix-schwarz code review findings are fixed and you can have a look again. Thanks!

**Comment by felix-schwarz** [comment]
@hosy Thanks for making the changes!

I noticed that now the passcode view in the Files.app now read `ownCloud File Provider UI` and went looking in the Info.plists for the other extensions to see how we solved it there. I then saw that `ownCloud` was hardcoded there and remember that we had similar issues in the past in those other targets, because `PRODUCT_NAME` is usually the name of the target, not `ownCloud`.

I assume the branding scripts then replace those hardcoded product names in the Info.plists, but also wondered if it's possible to define a project-wide "global" variable `APP_PRODUCT_NAME` - and use that in all Info.plists instead. 

I tried and it worked, so since I already had made the change, I committed it to fix the naming issue.

**Review by felix-schwarz — APPROVED** [review]

**Comment by jesmrec** [comment]
QA checks

- [x] Passcode enabled -> asked in Files App
- [X] Passcode + Biometrical enabled -> asked in Files App
- [X] Passcode disabled -> not asked in Files App
- [X] Lock disabled -> Passcode asked instantly
- [ ] Lock 1 min -> Passcode after 1 min in Files App

**Comment by jesmrec** [comment]
### (1)

Probably not a bug, but i want to be sure:

The passcode view that you see in Files App is always "Dark", even when you have a "Light" theme in the app. Are those views linked in some way? are they independent? is that expected?

**Comment by jesmrec** [comment]
### (2)

Check these steps out:

1. In the app set a passcode
2. Enter in Files App -> Passcode is asked
3. Enter a wrong password many times

Current: You can enter the wrong passcode as many times as you want, as fast as you can. If you close Files App and open it again, you will see the Passcode view but the buttons do not work (app is locked). Nothing indicates what's going on.

Expected: Same as the app: every time you fail, you will have to wait more time to do a new attempt (as in the app). In case the app is locked after several wrong passcodes, it should be indicated to the user somehow.

iPhone XR v14.4
`27ef6c99`

**Comment by hosy** [comment]
@jesmrec (1) and (2) are fixed.
(2) was a problem, which is currently also available in the share sheet in the release version.

**Comment by hosy** [comment]
@jesmrec please test this on iPad too.

**Comment by jesmrec** [comment]
### (3)

I have noticed the following behaviour in iPad:

1. In app, set passcode
2. Open Files App, ownCloud account
3. Passcode required, enter passcode

Current behaviour: passcode view is dismissed, ownCloud app is visible:


https://user-images.githubusercontent.com/14894746/110944280-8b1ad500-833c-11eb-939c-6345bce2d7b8.mov


Expected: When passcode view is dismissed, ownCloud app is visible

iPadAir2 v13.4
`c5fda239`

**Comment by hosy** [comment]
@jesmrec please deinstall the app and reinstall it again.it looks like the extension was not loaded.you can also try to￼ force quite files.app.

**Comment by jesmrec** [comment]
> @jesmrec please deinstall the app and reinstall it again.it looks like the extension was not loaded.you can also try to￼ force quite files.app.

It did not fix the problem. iPad is iOS13, could it affect?

**Comment by hosy** [comment]
@jesmrec FileProvider UI deployment target was set to iOS 14.2, now the minimum iOS version is 12.1. You can check it again!

**Comment by jesmrec** [comment]
(3) is fixed but ...

### (4)

with the iOS 13 iPad:

1. In app, set passcode and enable biometrical (touch/face)
2. Open Files App, ownCloud account
3. Enter touch id. 

Current: Files app is not unlocked

https://user-images.githubusercontent.com/14894746/111138419-3071bb00-8580-11eb-8be5-367978cc5296.MP4

Expected: Files app unlocked

iPadAir2 v13.4
`187e9914`

**Comment by hosy** [comment]
@jesmrec good catch! Should be fixed now with commit 10f74e7

**Comment by jesmrec** [comment]
### (5)

With commit `a9197e8` (sdk `29c2eb83`) , i can reproduce (3) in iPhone, after it was fixed in iPad

https://user-images.githubusercontent.com/14894746/111276734-f237d280-8637-11eb-94cb-dbc2608a8113.MP4

https://user-images.githubusercontent.com/14894746/111276721-ed731e80-8637-11eb-94e1-c00400115a03.MP4

**Comment by jesmrec** [comment]
This is approved on my side

**Review by jesmrec — APPROVED** [review]

**Pull request merged**